Mathlib/Topology/ContinuousFunction/CompactlySupported.lean

```diff
@@ -0,0 +1,700 @@
+        simp only
+        exact IsCompact.of_isClosed_subset isCompact_univ (isClosed_tsupport f)
+          (Set.subset_univ (tsupport f))
+        }
+  invFun f := f
+  left_inv f := by
+    ext
+    rfl
+  right_inv f := by
+    ext
+    rfl
+
+/-- A continuous function on a compact space has automatically compact support. This is not an
+instance to avoid type class loops. -/
+lemma compactlySupportedContinuousMapClass.ofCompact {G : Type*} [FunLike G α β]
+    [ContinuousMapClass G α β] [CompactSpace α] : CompactlySupportedContinuousMapClass G α β where
+  map_continuous := map_continuous
+  has_compact_support := by
+    intro f
+    exact IsCompact.of_isClosed_subset isCompact_univ (isClosed_tsupport f)
+      (Set.subset_univ (tsupport f))
+
+end Basics
+
+/-! ### Algebraic structure
+
+Whenever `β` has the structore of continuous addtive monoid and a compatible topological structure,
+then `C_c(α, β)` inherits a corresponding algebraic structure. The primary exception to this is that
+`C_c(α, β)` will not have a multiplicative identity.
+-/
+
+section AlgebraicStructure
+
+variable [TopologicalSpace β] (x : α)
+
+instance instZero [Zero β] : Zero C_c(α, β) where
+  zero := { toFun := (0 : C(α, β))
+            continuous_toFun := (0 : C(α, β)).2
+            has_compact_support' := by
+              rw [HasCompactSupport, tsupport]
+              simp only [ContinuousMap.coe_zero, Function.support_zero', closure_empty,
+                isCompact_empty]
+}
+
+instance instInhabited [Zero β] : Inhabited C_c(α, β) :=
+  ⟨0⟩
+
+@[simp]
+theorem coe_zero [Zero β] : ⇑(0 : C_c(α, β)) = 0 :=
+  rfl
+
+theorem zero_apply [Zero β] : (0 : C_c(α, β)) x = 0 :=
+  rfl
+
+instance instMul [MulZeroClass β] [ContinuousMul β] : Mul C_c(α, β) :=
+  ⟨fun f g => ⟨f * g, HasCompactSupport.mul_left g.2⟩⟩
+
+@[simp]
+theorem coe_mul [MulZeroClass β] [ContinuousMul β] (f g : C_c(α, β)) : ⇑(f * g) = f * g :=
+  rfl
+
+@[simp]
+theorem mul_apply [MulZeroClass β] [ContinuousMul β] (f g : C_c(α, β)) : (f * g) x = f x * g x :=
+  rfl
+
+-- the product of `(f : C(α, β))` and `(g : C_c(α, β))` is in `C_c(α, β)`
+instance instSMulC [MulZeroClass β] [ContinuousMul β] : SMul C(α, β) C_c(α, β) :=
+  ⟨fun f g => ⟨f * g, HasCompactSupport.mul_left g.2⟩⟩
+
+@[simp]
+theorem coe_smulc [MulZeroClass β] [ContinuousMul β] (f : C(α, β)) (g : C_c(α, β)) :
+    ⇑(f • g) = f * g :=
+  rfl
+
+@[simp]
+theorem smulc_apply [MulZeroClass β] [ContinuousMul β] (f : C(α, β)) (g : C_c(α, β)) :
+    (f • g) x = f x * g x :=
+  rfl
+
+instance instMulZeroClass [MulZeroClass β] [ContinuousMul β] : MulZeroClass C_c(α, β) :=
+  DFunLike.coe_injective.mulZeroClass _ coe_zero coe_mul
+
+instance instSemigroupWithZero [SemigroupWithZero β] [ContinuousMul β] :
+    SemigroupWithZero C_c(α, β) :=
+  DFunLike.coe_injective.semigroupWithZero _ coe_zero coe_mul
+
+-- need `[AddMonoid β]` here to apply `HasCompactSupport.add`
+instance instAdd [AddMonoid β] [ContinuousAdd β] : Add C_c(α, β) :=
```

Comment:
A `AddZeroClass` should be enough, here ane below. For this, can you fix the definition of `HasCompactMulSupport.mul`, requiring only `MulOneClass β`?

Reply:
OK, I will edit Topology/Support.